src/GraphQLController.php

```diff
@@ -37,9 +37,19 @@ public function query(Request $request, string $schema = null): JsonResponse
             $schema = config('graphql.default_schema');
         }
 
-        // If a singular query was not found, it means the queries are in batch
-        $isBatch = ! $request->has('query');
-        $inputs = $isBatch ? $request->input() : [$request->input()];
+        // If a singular query parameter is not found, assume a batch request
+        $isBatch = false;
+        if ($request->input()) {
+            if ($request->has('query')) {
+                $inputs = [$request->input()];
+            } else {
+                $isBatch = true;
+                $inputs = $request->input();
+            }
+        } else {
+            // The request input is completely empty, assemble inputs accordingly
+            $inputs = [[]];
+        }
```

Comment:
Somehow I get the feeling this is overdoing things here.

When I revert this block back to the original version, only one test fails => the _first_ one in `\Rebing\GraphQL\Tests\Database\EmptyQueryTest::dataForEmptyQuery`:
```php
            // single request which is completely empty
            [
                false, [],
            ],
```
A request which is completely empty is, due to the current logic, expected to be a batch request. Yet one without any actual batches.

So the expected result to me is an empty array and not a syntax error.

Can you explain your goal of the new changes here?

Reply:
> A request which is completely empty is, due to the current logic, expected to be a batch request. Yet one without any actual batches.

What I can imagine is that, if we would have the ability to _disable_ batch requests, _then_ being extra strict here makes sense.

That's a different change/PR though but one I could also get on board, as batch requests could have unforeseen consequences and security implications.

Reply:
Addressing this incrementally indeed makes sense: The initial intention here was after all to resolve the hard exception. I have prepared a new branch based on 5.x, with the reverted changes. Let me know if you'd like me to submit that as a new PR:
https://github.com/rebing/graphql-laravel/compare/5.x...plivius:feature/proper-syntax-error-for-empty-query-5x

In the near future, it might still be a good idea to look into the case of the empty request (indeed perhaps dependent on enable/disable batch requests). Though the spec doesn't appear to be really explicit on this, if empty single/batch queries are considered incorrect, it would make sense the same is true for completely empty requests.

Reply:
Hi @mfn, I've created a separate PR (#685) based on `5.x`, let me know if you would like to see any changes.

Reply:
Thank you, I've noticed it already but currently lack enough time.